net/freeradius/src/opnsense/service/templates/OPNsense/Freeradius/proxy.conf

```
@@ -0,0 +1,128 @@
+{% if helpers.exists('OPNsense.freeradius.general.fallbackproxy') and OPNsense.freeradius.general.fallbackproxy == '1' %}
+proxy server {
+        default_fallback = yes
+}
+{% else %}
+proxy server {
+        default_fallback = no
+}
+{% endif %}
```

why not do this? in general I think this will be a maintenance nightmare going forward. most of the controller code is very likely spurious and the line ending changes aren't my favourite thing to see, especially in files that have no changes.

```suggestion
proxy server {
{% if helpers.exists('OPNsense.freeradius.general.fallbackproxy') and OPNsense.freeradius.general.fallbackproxy == '1' %}
        default_fallback = yes
{% else %}
        default_fallback = no
{% endif %}
}
```

---

Thanks for the suggestion, how can I optimize the spurious code?

---

[core.git/src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php has more](https://github.com/opnsense/core/blob/master/src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php) offers getAction() and setAction() already as well as other helpers and shortcuts to minimise API controller code. I guess this was copied from somewhere that had the code, but this way it just gets copied and hardcoded over and over (take a look at all the validation code duplication instead of using the respective validateAndSave() alternatives...).

---

Thank you I'll study the other helpers for the next time to try to optimize my code.